tests/cpp/nodes/test_collections.cpp

```diff
@@ -463,6 +450,8 @@ TEST_CASE("SetNode") {
 
         auto ptr = graph.emplace_node<SetNode>(num_elements);
 
+        graph.emplace_node<ArrayValidationNode>(ptr);
```

Comment:
I think you're still missing propagate for this one

Reply:
Good catch. Ok, this time I think I got them all.

Reply:
ah, still an error. One sec.

Reply:
ok, fixed. This time for real.